### PR TITLE
fix(lr-scheduler): add argument to plateau, infer steps for 1cycle, add more tests

### PR DESCRIPTION
**Pull request by cxzhang4**

### CODE REVIEW

File: R/CallbackSetLRScheduler.R [hunk 1]
```diff
@@ -41,15 +41,31 @@ CallbackSetLRScheduler = R6Class("CallbackSetLRScheduler",
 
       self$scheduler_fn = .scheduler
       private$.scheduler_args = list(...)
+
       if (step_on_epoch) {
-        self$on_epoch_end = function() self$scheduler$step()
+        if (class(self$scheduler_fn)[[1L]] == "lr_reduce_on_plateau") {
```
Comment (sebffischer) [inline R/CallbackSetLRScheduler.R:46]: I would suggest we solve this differently.
I would probably just implement a custom sub-class for the `lr_reduce_on_plateu` scheduler.

**Review by sebffischer — CHANGES_REQUESTED** [review]

File: tests/testthat/helper_autotest.R [hunk 2]
```diff
@@ -307,10 +307,19 @@ expect_torch_callback = function(torch_callback, check_man = TRUE, check_paramse
     paramtest = expect_paramset(torch_callback$param_set, init_fn)
     expect_paramtest(paramtest)
   }
-  implemented_stages = names(cbgen$public_methods)[grepl("^on_", names(cbgen$public_methods))]
+
+  super_cbgen_public_methods = cbgen$get_inherit()$public_methods 
+  superclass_implemented_stages = names(super_cbgen_public_methods)[grepl("^on_", names(super_cbgen_public_methods))]
+  implemented_stages = union(superclass_implemented_stages, names(cbgen$public_methods)[grepl("^on_", names(cbgen$public_methods))])
```
Comment (cxzhang4) [inline tests/testthat/helper_autotest.R:313]: I think it makes sense to check the superclass for implemented stages, since a subclass would inherit any stages that it doesn't override.

File: tests/testthat/test_CallbackSetLRScheduler.R [hunk 3]
```diff
@@ -44,6 +109,45 @@ test_that("decay works", {
...
+  task = tsk("iris")
+
+  mlp = lrn("classif.mlp",
+    callbacks = cb,
+    epochs = 10, batch_size = 50, neurons = 10,
+    measures_train = msrs(c("classif.acc", "classif.ce"))
+  )
+
+  mlp$train(task)
+
```
Comment (cxzhang4) [inline tests/testthat/test_CallbackSetLRScheduler.R:146]: Do we need a "proper" expectation here for the final learning rate? I suspect it's okay not to have one.

I was struggling a bit with the calculations, and LLMs consistently produce values that don't match the final learning rate.

File: NEWS.md [hunk 4]
```diff
@@ -27,6 +27,8 @@
 * feat: Training now does not fail anymore when encountering a missing value
   during validation but uses `NA` instead.
 * feat: The context for callbacks now includes the network prediction (`y_hat`).
+* fix: The `lr_reduce_on_plateau` callback now works.
+* feat: The `lr_one_cycle` callback now infers the total number of steps.
```
Comment (sebffischer) [inline NEWS.md:31]: are there any other schedulers that have an argument like this?

File: R/CallbackSetLRScheduler.R [hunk 5]
```diff
@@ -33,6 +33,8 @@ CallbackSetLRScheduler = R6Class("CallbackSetLRScheduler",
     scheduler = NULL,
     #' @description
     #' Creates a new instance of this [R6][R6::R6Class] class.
+    #' @param .scheduler (`LRScheduler`)\cr
```
Comment (sebffischer) [inline R/CallbackSetLRScheduler.R:36]: already documented above

File: R/CallbackSetLRScheduler.R [hunk 6]
```diff
@@ -58,6 +61,45 @@ CallbackSetLRScheduler = R6Class("CallbackSetLRScheduler",
   )
 )
 
+CallbackSetLRSchedulerOneCycle = R6Class("CallbackSetLRSchedulerOneCycle",
```
Comment (sebffischer) [inline R/CallbackSetLRScheduler.R:64]: Needs documentation

File: R/CallbackSetLRScheduler.R [hunk 7]
```diff
@@ -185,6 +227,8 @@ mlr3torch_callbacks$add("lr_step", function() {
 #'   The `torch` scheduler generator defined using `torch::lr_scheduler()`.
 #' @param step_on_epoch (`logical(1)`)\cr
 #'   Whether the scheduler steps after every epoch
+#' @param step_takes_valid_metric (`logical(1)`)\cr
```
Comment (sebffischer) [inline R/CallbackSetLRScheduler.R:230]: leftover

File: tests/testthat/helper_autotest.R (hunk 2 above)
Comment (sebffischer) [reply tests/testthat/helper_autotest.R:313]: do it recursively.

File: tests/testthat/test_CallbackSetLRScheduler.R (hunk 3 above)
Comment (sebffischer) [reply tests/testthat/test_CallbackSetLRScheduler.R:146]: don't need to do this.

**Review by sebffischer — CHANGES_REQUESTED** [review]
